lib/mix/lib/mix/config.ex

```diff
@@ -146,9 +146,10 @@ defmodule Mix.Config do
   Reads many configuration files given by wildcard into a single config.
   """
   def read_wildcard!(config, path) do
-    paths = case Path.wildcard(path) do
-      [] -> [path]
-      o  -> o
+    paths = if path =~ ~r/[*?[{]/ do
```

Comment:
Also, should we add some code comments here explaining that we are checking for the path because we would like to raise for missing files?